core/templates/dev/head/pages/exploration_editor/simple_editor_tab/SimpleEditorManagerServiceSpec.js

```diff
@@ -0,0 +1,344 @@
+    // data, initialized in beforeEach block and StatesToQuestionsService
+    // .getQuestions() will return non null value.
+    expectedReturnedValue = simpleEditorManagerService.tryToInit();
+    expect(simpleEditorManagerService.getTitle()).toBe(data.title);
+    expect(simpleEditorManagerService.getIntroductionHtml())
+      .toBe(data.states.Introduction.content[0].value);
+    // Expected questions are genereated from test data.
+
+    var expectedSimpleEditorManagerQuestionList = questionList;
+    expect(simpleEditorManagerService.getQuestionList())
+      .toEqual(expectedSimpleEditorManagerQuestionList);
+    expect(expectedReturnedValue).toBeTruthy();
+
+    // Case:2 when StatesToQuestionsService.getQuestions() will return null,
+    // may be due to errors.
+    spyOn(statesToQuestionsService, 'getQuestions').andReturn(null);
+    expectedReturnedValue = simpleEditorManagerService.tryToInit();
+    expect(expectedReturnedValue).toBeFalsy();
+  });
+
+  it('should return data', function() {
+    // Expected initialized data inside simpleEditorManagerService
+    var expectedSimpleEditorManagerData = {
+      title: data.title,
+      introductionHtml: data.states.Introduction.content[0].value,
+      questionList: null
+    };
+    // Initialize simpleEditorManagerService.data
+    simpleEditorManagerService.tryToInit();
+    expectedSimpleEditorManagerData.questionList = questionList;
+    expect(simpleEditorManagerService.getData())
+      .toEqual(expectedSimpleEditorManagerData);
+  });
+
+  it('should return title', function() {
+    var expectedTitle = data.title;
+    // Initialize simpleEditorManagerService.data
+    simpleEditorManagerService.tryToInit();
+    expect(simpleEditorManagerService.getTitle()).toEqual(expectedTitle);
+  });
+
+  it('should return introduction html', function() {
+    var expectedIntroductionHtml = data.states.Introduction.content[0].value;
+    simpleEditorManagerService.tryToInit();
+    expect(simpleEditorManagerService.getIntroductionHtml())
+      .toEqual(expectedIntroductionHtml);
+  });
+
+  it('should return question list', function() {
+    var expectedSimpleEditorManagerQuestionList = questionList;
+    simpleEditorManagerService.tryToInit();
+    expect(simpleEditorManagerService.getQuestionList())
+      .toEqual(expectedSimpleEditorManagerQuestionList);
+  });
+
+  it('should save the given new title', function() {
+    var expectedNewTitle = 'ExpNewTitle';
+    simpleEditorManagerService.saveTitle(expectedNewTitle);
+    expect(explorationTitleService.savedMemento).toEqual(expectedNewTitle);
+    expect(simpleEditorManagerService.getTitle()).toEqual(expectedNewTitle);
+  });
+
+  it('should save the given new introduction html', function() {
+    var expectedNewIntroductionHtmll = '<p> new intro </p>';
+    simpleEditorManagerService
+      .saveIntroductionHtml(expectedNewIntroductionHtmll);
+    expect(simpleEditorShimService.getIntroductionHtml())
```

Comment:
Do you need this here? I thought this test suite was testing simpleEditorManagerService